### PR TITLE
fix(torch): display msg if resume fails, also fails if not best_model…

### PR DESCRIPTION
**Pull request by fantes**
….txt file

### CODE REVIEW

File: src/backends/torch/torchsolver.cc [hunk 1]
```diff
@@ -277,6 +277,25 @@ namespace dd
     // reload solver if asked for and set it value accordingly
     if (ad_mllib.has("resume") && ad_mllib.get("resume").get<bool>())
       {
+        if (mlmodel._sstate.empty())
+          {
+            std::string msg
+                = "resuming a model requires a solverstate file in model "
```
Comment (beniz) [inline src/backends/torch/torchsolver.cc:283]: `solverstate` was the file extension for caffe, I suggest we change it here so it is relevant for torch.

File: src/backends/torch/torchsolver.cc [hunk 2]
```diff
@@ -277,6 +277,25 @@ namespace dd
...
+            throw MLLibBadParamException(msg);
+          }
+        else
+          try
+            {
+              return load(mlmodel._sstate, main_device);
+            }
+          catch (std::exception &e)
+            {
+              this->_logger->error("Failed to load solver state");
```
Comment (beniz) [inline src/backends/torch/torchsolver.cc:295]: maybe log the name of the directory in which the file should be found ?

File: src/backends/torch/torchsolver.cc [hunk 3]
```diff
@@ -291,6 +310,14 @@ namespace dd
                 std::string bin;
                 std::string test_name;
                 bestfile.open(bestfilename, std::ios::in);
+                if (!bestfile.is_open())
```
Comment (beniz) [inline src/backends/torch/torchsolver.cc:313]: Does this mean resuming fails if there's a file but no identified best model ?

File: src/backends/torch/torchsolver.cc [hunk 4]
```diff
@@ -306,30 +333,15 @@ namespace dd
...
-              this->_logger->error("Failed to load solver state");
-              throw;
-            }
       }
     else if (!mlmodel._sstate.empty())
       {
         this->_logger->error("not resuming while a solverstate file remains "
                              "in model repository");
         throw MLLibBadParamException(
             "a solverstate file requires a resume argument for training, "
```
Comment (beniz) [inline src/backends/torch/torchsolver.cc:342]: Same remark as above, we should swap `solverstate` with another formulation that is better grounded to torch filenames.

File: src/backends/torch/torchsolver.cc (hunk 1 above)
Comment (fantes) [reply src/backends/torch/torchsolver.cc:283]: ok, done

File: src/backends/torch/torchsolver.cc (hunk 2 above)
Comment (fantes) [reply src/backends/torch/torchsolver.cc:295]: ok

File: src/backends/torch/torchsolver.cc (hunk 3 above)
Comment (fantes) [reply src/backends/torch/torchsolver.cc:313]: yes, and it is corrected now: resume can be done even if not test has been done yet

**Review by Bycob — APPROVED** [review]

**Review by beniz — APPROVED** [review]

**Pull request merged**